tensorboard/webapp/customization/customizable_component.ts

```diff
@@ -0,0 +1,102 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {
+  Component,
+  ComponentFactoryResolver,
+  Input,
+  OnInit,
+  Type,
+  ViewContainerRef,
+} from '@angular/core';
+
+/**
+ * A Component that defines a customization point. Ideal for use for small
+ * surgical customizations deep within the component hierarchy. The
+ * customizations can be provided dynamically via Angular dependency injection
+ * and providers.
+ *
+ * Cookbook:
+ *
+ * 1. Define an empty marker class that a Component will have to implement in
+ *    order to be recognized as the customization:
+ *
+ *    export class SomeCustomComponentType {};
```

Comment:
Perhaps update the cookbook to add below instead of declaring a class:

```ts
 const CustomizableButton = new InjectionToken<Type<unknown>>('Customizable Button');
```

Reply:
Done.